storage/types.go

```diff
@@ -93,16 +93,18 @@ func NewNodeIDForTreeCoords(depth int64, index int64, maxLenBits int) NodeID {
 
 // SetBit sets the ith bit to true if b is non-zero, and false otherwise.
 func (n *NodeID) SetBit(i int, b uint) {
+	bIndex := (n.PathLenBits - i - 1) / 8
 	if b == 0 {
-		n.Path[i/8] &= ^(1 << uint(i%8))
+		n.Path[bIndex] &= ^(1 << uint(i%8))
 	} else {
-		n.Path[i/8] |= (1 << uint(i%8))
+		n.Path[bIndex] |= (1 << uint(i%8))
 	}
 }
 
 // Bit returns 1 if the ith bit is true, and false otherwise.
 func (n *NodeID) Bit(i int) uint {
-	return uint((n.Path[i/8] >> uint(i%8)) & 0x01)
+	bIndex := (n.PathLenBits - i - 1) / 8
```

Comment:
Doc comment should probably mention the endianicity, endianity ... which way round it is,